src/wallet/wallet.cpp

```diff
@@ -2242,6 +2244,8 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
                 if (coinControl && nFeeNeeded > 0 && coinControl->nMinimumTotalFee > nFeeNeeded) {
                     nFeeNeeded = coinControl->nMinimumTotalFee;
                 }
+                if (coinControl && coinControl->nFeeRate > CFeeRate(0))
+                    nFeeNeeded = coinControl->nFeeRate.GetFee(nBytes);
```

Comment:
Yes. We could use an extra boolean (like `overrideFeeRate`). My thoughts where that a `CFeeRate(0)` is not practical but I agree, it would technically be possible to use a zero feerate. I'll add the bool.